### DISCUSSION TITLE
LDAP default user role && LDAP conflict to already exist users

### DISCUSSION THREAD

**Discussion by lpb1** [open-webui/open-webui#6943] [Issues]
# Bug Report

## Installation Method

1. Reproduce the production environment
docker run -d \
    -p 8082:8080 \
    -e “OPENAI_API_BASE_URLS=http://xxxx;https://xxxx” \
    -e “OPENAI_API_KEYS=sk-test;sk-test” \
    -e OLLAMA_BASE_URL=http://xxxx:11434 \
    -v /open-webui/data:/app/backend/data \
    --name open-webui \
    --restart always \
    ghcr.io/open-webui/open-webui:v0.3.34
2. Upgrade to the version that includes LDAP
docker compose.yaml:
version: '3'
services:
  openwebui:
    image: ghcr.io/open-webui/open-webui:git-f5f2215
    container_name: open-webui
    ports:
      - "3000:8080"
    volumes:
      - /open-webui/data:/app/backend/data
    environment:
      - 'GLOBAL_LOG_LEVEL=DEBUG'
      - 'OLLAMA_BASE_URL=http://xxxx:11434'
      - 'OPENAI_API_BASE_URLS=http://xxxx;https://xxxx'
      - "OPENAI_API_KEYS=sk-test;sk-test"
      - 'ENABLE_LDAP=True'
      - 'LDAP_SERVER_HOST=xxxx'
      - 'LDAP_SERVER_PORT=xxxx'
      - "LDAP_APP_PASSWORD=$LDAP_APP_PASSWORD"
      - 'LDAP_USE_TLS=False'
      - 'LDAP_APP_DN=cn=LDAP User,cn=Users,dc=xxxx,dc=xxxx'
      - 'LDAP_SEARCH_BASE=dc=xxxx,dc=xxxx'
      - 'LDAP_SEARCH_FILTERS=(mail=%s)'
      - 'LDAP_ATTRIBUTE_FOR_USERNAME=mail'

## Environment

- **Open WebUI Version:** [git-f5f2215]

- **Operating System:** [Ubuntu 22.04]

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

If the email is not registered before LDAP integrated, LDAP user can login directly, if the setting of default user role is USER

If the email is registered before LDAP integrated, LDAP account should be a new user or inherit from already exist local account.

## Actual Behavior:

The email is not registered before LDAP integrated:
LDAP user can not login directly, the status is pending, even the setting of default user role is USER.

The email is registered before LDAP integrated:
LDAP user login failed both with domain password and already registered password, already registered local user login successful with data

## Description

**Bug Summary:**
already exist same name local account conflict to ldap accounts

## Reproduction Details

**Steps to Reproduce:**
[Outline the steps to reproduce the bug. Be as detailed as possible.]

## Logs and Screenshots

**Browser Console Logs:**
Failed to load resource: the server responded with a status of 400 (Bad Request)
index.ts:132 Objectdetail: "400: The email or password provided is incorrect. Please check for typos and try logging in again."[[Prototype]]: Objectconstructor: ƒ Object()hasOwnProperty: ƒ hasOwnProperty()isPrototypeOf: ƒ isPrototypeOf()propertyIsEnumerable: ƒ propertyIsEnumerable()toLocaleString: ƒ toLocaleString()toString: ƒ toString()valueOf: ƒ valueOf()__defineGetter__: ƒ __defineGetter__()__defineSetter__: ƒ __defineSetter__()__lookupGetter__: ƒ __lookupGetter__()__lookupSetter__: ƒ __lookupSetter__()__proto__: (...)get __proto__: ƒ __proto__()set __proto__: ƒ __proto__()
api/v1/auths/ldap:1 
        

**Docker Container Logs:**
INFO  [open_webui.apps.webui.models.auths] authenticate_user: pblv@xxxx.com
DEBUG [open_webui.main] Commit session after request
INFO:     xxxx:61395 - "POST /api/v1/auths/ldap HTTP/1.1" 400 Bad Request

**Screenshots/Screen Recordings (if applicable):**
<img width="626" alt="image" src="https://github.com/user-attachments/assets/262026ed-ff8f-4a3f-8bac-d82328f18e69">

**Comment by stondino00** [comment]
Getting this same now after the 0.4.0 ldap feature release.

**Comment by foxworld306** [comment]
This post resolved my confusion after upgrading to 0.4.0/0.4.1. I indeed encountered the same issue, where I had to manually change the status of users who logged in for the first time using LDAP from 'pending' to 'user,' even though I had already specified this behavior in the settings. Additionally, I am unable to use email addresses that were registered before enabling LDAP authentication.
![image](https://github.com/user-attachments/assets/ec8a09ff-e9eb-4bae-ba5b-27e790ac0244)
![image](https://github.com/user-attachments/assets/3e13d96d-f16f-4643-a585-005f00210eec)

**Reply by strobel1x** [reply]
The user needs to set the password in open-webui to the one used for LDAP authentification, then the LDAP auth for the exiting user works here.

**Reply by foxworld306** [reply]
Thanks a lot! It works after reset existing users' password.

**Comment by guijarro** [comment]
Is it possible to configure OpenWEbUI so that it sees all groups which are defined in my LDAP server? This will save me from having to build groups with the OpenWebUI interface. Please let me know if I have to place this question somewhere else.
